tests/nativeapp/codegen/snowpark/test_python_processor.py

```diff
@@ -353,7 +354,17 @@ def test_process_no_collected_functions(
                 project_definition=native_app_project_instance.native_app,
                 project_root=local_path,
             )
-            SnowparkAnnotationProcessor(na_project=project).process(
+            processor = SnowparkAnnotationProcessor(
+                BundleContext(
+                    package_name=project.package_name,
+                    artifacts=project.artifacts,
+                    project_root=project.project_root,
+                    deploy_root=project.deploy_root,
+                    bundle_root=project.bundle_root,
+                    generated_root=project.generated_root,
+                )
```

Comment:
Could introduce static `from_` methods on the `BundleContext` dataclass to prevent duplication / errors

Reply:
Good suggestion! Added `from_na_project()`